wandb/sdk/launch/utils.py

```diff
@@ -180,6 +181,26 @@ def fetch_wandb_project_run_info(uri, api=None):
     return result
 
 
+def fetch_project_diff(uri, api=None):
+    patch = None
+    try:
+        entity, project, name = parse_wandb_uri(uri)
+        (_, _, patch, _) = api.run_config(project, name, entity)
+    except requests.exceptions.HTTPError:
+        pass
+    return patch
+
+
+def apply_patch(patch_string, dst_dir):
+    with open(os.path.join(dst_dir, "diff.patch"), "w") as fp:
+        fp.write(patch_string)
+    os.system(
```

Comment:
Would this be better to do with a subprocess and check that it worked then warn if it failed?

Reply:
subprocess.check_call() seems like a better option.